File: python/tests/test_garudadidada.py
from garudadidada import format_bytes, format_duration, ordinal, pluralize


def test_format_bytes():
    assert format_bytes(0) == "0 B"
    assert format_bytes(500) == "500 B"
    assert format_bytes(1024) == "1 KB"
    assert format_bytes(1536) == "1.5 KB"
    assert format_bytes(1048576) == "1 MB"
    assert format_bytes(1073741824) == "1 GB"


def test_format_duration():
    assert format_duration(0) == "0s"
    assert format_duration(45) == "45s"
    assert format_duration(90) == "1m 30s"
    assert format_duration(3661) == "1h 1m 1s"
    assert format_duration(90061) == "1d 1h 1m 1s"


def test_ordinal():
    assert ordinal(1) == "1st"
    assert ordinal(2) == "2nd"
    assert ordinal(3) == "3rd"
    assert ordinal(4) == "4th"
    assert ordinal(11) == "11th"
    assert ordinal(12) == "12th"
    assert ordinal(13) == "13th"
    assert ordinal(21) == "21st"
    assert ordinal(22) == "22nd"
    assert ordinal(101) == "101st"
    assert ordinal(111) == "111th"


def test_pluralize():
    assert pluralize(1, "file") == "1 file"
    assert pluralize(3, "file") == "3 files"
    assert pluralize(1, "file", "files") == "1 file"
    assert pluralize(2, "person", "people") == "2 people"

File: ruby/test/test_garudadidada.rb
# frozen_string_literal: true

require "minitest/autorun"
require_relative "../lib/garudadidada"

class TestGarudadidada < Minitest::Test
  def test_format_bytes
    assert_equal "0 B", Garudadidada.format_bytes(0)
    assert_equal "500 B", Garudadidada.format_bytes(500)
    assert_equal "1 KB", Garudadidada.format_bytes(1024)
    assert_equal "1.5 KB", Garudadidada.format_bytes(1536)
    assert_equal "1 MB", Garudadidada.format_bytes(1_048_576)
    assert_equal "1 GB", Garudadidada.format_bytes(1_073_741_824)
  end

  def test_format_duration
    assert_equal "0s", Garudadidada.format_duration(0)
    assert_equal "45s", Garudadidada.format_duration(45)
    assert_equal "1m 30s", Garudadidada.format_duration(90)
    assert_equal "1h 1m 1s", Garudadidada.format_duration(3661)
    assert_equal "1d 1h 1m 1s", Garudadidada.format_duration(90_061)
  end

  def test_ordinal
    assert_equal "1st", Garudadidada.ordinal(1)
    assert_equal "2nd", Garudadidada.ordinal(2)
    assert_equal "3rd", Garudadidada.ordinal(3)
    assert_equal "11th", Garudadidada.ordinal(11)
    assert_equal "12th", Garudadidada.ordinal(12)
    assert_equal "13th", Garudadidada.ordinal(13)
    assert_equal "21st", Garudadidada.ordinal(21)
    assert_equal "22nd", Garudadidada.ordinal(22)
    assert_equal "101st", Garudadidada.ordinal(101)
    assert_equal "111th", Garudadidada.ordinal(111)
  end

  def test_pluralize
    assert_equal "1 file", Garudadidada.pluralize(1, "file")
    assert_equal "3 files", Garudadidada.pluralize(3, "file")
    assert_equal "2 people", Garudadidada.pluralize(2, "person", "people")
  end
end
